### PR TITLE
Migrate from the OpenJDK legacy image to the newest image

### LINKED ISSUE #1685: Migrate from the OpenJDK legacy image to the newest image
Relevant deprecation post: https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034

According to CircleCI ops:
```
Migrating from the OpenJDK legacy image to the newest image is a small config change.

circleci/openjdk:11.0.0 -> cimg/openjdk:11.0.0
```

### PR DESCRIPTION
**Pull request by efeg**
This PR resolves #1685.

### CODE REVIEW

**Comment by efeg** [comment]
Looks like CircleCI has a docker image with a misconfigured environment variable `JAVA_HOME`.
The value `JAVA_HOME` is configured with was `/usr/local/jdk-11.0.0`, but the value it should have been configured with is `/usr/local/jdk-11`. I explicitly set this env. variable to the correct path to resolve the following (see [failed job](https://app.circleci.com/pipelines/github/linkedin/cruise-control/1425/workflows/4087d442-9945-4dc3-b659-f95c66db0eaa/jobs/4643)):
```
ERROR: JAVA_HOME is set to an invalid directory: /usr/local/jdk-11.0.0

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -4,10 +4,11 @@ jobs:
 
   build:
     environment:
+      JAVA_HOME: "/usr/local/jdk-11"
       _JAVA_OPTIONS: "-Xms512m -Xmx1g"
     working_directory: ~/workspace
     docker:
-      - image: circleci/openjdk:11-jdk
+      - image: cimg/openjdk:11.0.0
```
Comment (amuraru) [inline .circleci/config.yml:11]: should the latest `cimg/openjdk:11.0.12` be used here? Or maybe `cimg/openjdk:11.0` to always get the latest jdk 11 patch version?

Comment (efeg) [reply .circleci/config.yml:11]: Thanks @amuraru -- using minor release tag (i.e. `cimg/openjdk:11.0`) is a good idea. Looks like doing so will indeed automatically point to future patch updates as they are released by AdoptOpenJDK & Oracle (source: https://circleci.com/developer/images/image/cimg/openjdk)

Comment (efeg) [reply .circleci/config.yml]: Latest job gets java `11.0.12`. 
```
$ /usr/local/jdk-11.0.12/bin/java --version
Picked up _JAVA_OPTIONS: -Xms512m -Xmx1g
openjdk 11.0.12 2021-07-20
OpenJDK Runtime Environment Temurin-11.0.12+7 (build 11.0.12+7)
OpenJDK 64-Bit Server VM Temurin-11.0.12+7 (build 11.0.12+7, mixed mode)
```

**Review by zornhsu — APPROVED** [review]

**Pull request merged**